vendor/github.com/Microsoft/hcsshim/internal/safefile/safeopen.go

```diff
@@ -293,12 +293,12 @@ func RemoveAllRelative(path string, root *os.File) error {
 	for {
 		names, err1 := fd.Readdirnames(100)
 		for _, name := range names {
-			err1 := RemoveAllRelative(path+string(os.PathSeparator)+name, root)
-			if err == nil {
-				err = err1
+			if err2 := RemoveAllRelative(path+string(os.PathSeparator)+name, root); err == nil {
+				err = err2
```

Comment:
Was glancing over changes in the update, and not a fan of formatting this code like this; the previous formatting made it much less error-prone; 


- it makes it _really_ easy to read the code to be checking for `err2`, but it's not.
- it makes it _really_ easy to think it's checking for non-`nil` error, but it's not.

And the overall use of a `err` as global, reused variable is a real risk of that variable being accidentally shadowed. I wonder if this code can be rewritten with early returns or otherwise.

Reply:
Yup; was planning to open a ticket or PR (definitely not an issue with your PR, but I stumbled upon it)